src/core/inductive.ml

```diff
@@ -137,32 +205,28 @@ let gen_rec_type : Sig_state.t -> popt -> sym -> sym list -> term =
         Extra.StrSet.add s set
       else set
     in
-    List.fold_left f Extra.StrSet.empty (ind_sym::cons_list)
-  in
-  let x_str = Extra.get_safe_prefix "x" set in
+    let flat_cons_list = List.flatten cons_list_list in
+    List.fold_left f Extra.StrSet.empty (ind_typ_list @ flat_cons_list) in
```

Comment:
This could be improved:
```
let add_names_from_cons_list set l = List.fold_left add_name_from_sym set l in
let set = List.fold_left add_names_from_cons_list Extra.StrSet.empty cons_list_list in
List.fold_left add_name_from_sym set ind_typ_list
```

Reply:
You right yes, so done.